Files/ViewModels/ItemViewModel.cs

```diff
@@ -1171,28 +1254,34 @@ public async Task<bool> EnumerateItemsFromStandardFolderAsync(string path)
                             hasNextFile = FindNextFile(hFile, out findData);
                             if (count == 32 || count % 300 == 0 || !hasNextFile)
                             {
-                                var orderedList = OrderFiles2(tempList);
-                                await CoreApplication.MainView.ExecuteOnUIThreadAsync(() =>
-                                {
-                                    OrderFiles(orderedList);
-                                });
+                                await OrderFilesAndFoldersAsync();
+                                await ApplyFilesAndFoldersChangesAsync();
                             }
                         } while (hasNextFile);
+                    });
 
+                    if (!addFilesCTS.IsCancellationRequested)
+                    {
                         await fileListCache.SaveFileListToCache(path, new CacheEntry
                         {
                             CurrentFolder = CurrentFolder,
-                            FileList = tempList
+                            // since filesAndFolders could be mutated, memory cache needs a copy of current list
+                            FileList = filesAndFolders.ToArray().ToList()
```

Comment:
'ToList()' on a list will create copy too, no need for double allocation here

Reply:
Fixed via c70cf14

Reply:
@hez2010 I think some rebase reverted that change

Reply:
Fixed via 77503ed